rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/learning/sync/listener/SyncTrainingListener.java

```diff
@@ -0,0 +1,31 @@
+package org.deeplearning4j.rl4j.learning.sync.listener;
+
+import org.deeplearning4j.rl4j.util.IDataManager;
+
+/**
+ * A listener interface to use with a descendant of {@link org.deeplearning4j.rl4j.learning.sync.SyncLearning}
+ */
+public interface SyncTrainingListener {
+    /**
+     * Called once when the training starts. The training can be aborted by calling event.setCanContinue(false)
```

Comment:
@AlexDBlack Are you OK with using a listener for that?

Reply:
Yes, we're actually looking at a similar design in SameDiff listeners.
Though we're going with something like onEpochEnd returning an enumeration - `ListenerResponse.STOP` or `ListenerResponse.CONTINUE` sort of thing.